store/types/listening.go

```diff
@@ -0,0 +1,42 @@
+package types
+
+import (
+	"io"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+)
+
+// WriteListener interface for streaming data out from a listenkv.Store
+type WriteListener interface {
+	// if value is nil then it was deleted
+	// storeKey indicates the source KVStore, to facilitate using the the same WriteListener across separate KVStores
+	// set bool indicates if it was a set; true: set, false: delete
+	OnWrite(storeKey StoreKey, set bool, key []byte, value []byte)
+}
+
+// StoreKVPairWriteListener is used to configure listening to a KVStore by writing out length-prefixed
+// protobuf encoded StoreKVPairs to an underlying io.Writer
+type StoreKVPairWriteListener struct {
+	writer     io.Writer
+	marshaller codec.BinaryMarshaler
+}
+
+// NewStoreKVPairWriteListener wraps creates a StoreKVPairWriteListener with a provdied io.Writer and codec.BinaryMarshaler
+func NewStoreKVPairWriteListener(w io.Writer, m codec.BinaryMarshaler) *StoreKVPairWriteListener {
+	return &StoreKVPairWriteListener{
+		writer:     w,
+		marshaller: m,
+	}
+}
+
+// OnWrite satisfies the WriteListener interface by writing length-prefixed protobuf encoded StoreKVPairs
+func (wl *StoreKVPairWriteListener) OnWrite(storeKey StoreKey, set bool, key []byte, value []byte) {
```

Comment:
The error is silently ignored here. Would it make sense to update the `OnWrite` interface to add an error? The listener then decides what to do on error.

Reply:
Yeah I think that makes sense! `OnWrite` is only called from within the listen.KvStore, we can return it to be handled there but the methods it is called within (`Set` and `Delete`) don't return an error themselves so it can't bubble up any further than that.

Reply:
My only hesitation is that there doesn't seem to be much precedent for initializing a KVStore with a logger to log from within that level.

Reply:
Ah, I see, so `Set and Delete` will anyways silently ignore those errors.

Yeah introducing a logger there is a bigger refactor. I still think adding `error` on OnWrite is useful, but if you think it's superfluous we can remove it too.

Reply:
I went ahead and added it to `OnWrite`! That way we don't have to break the interface if we want to log down the road.